drivers/storage/portworx/status.go

```diff
@@ -195,6 +212,99 @@ func (p *portworx) updatePortworxMigrationStatus(
 	}
 }
 
+// updatePortworxInstallStatus updates portworx install status, expected status: InProgress, Completed
+// If there's 1 node has Initializing status and portworx is not initialized, mark as PortworxInstallInProgress
+// If all nodes status are updated according to SDK server response and portworx is installing, mark as PortworxInstallCompleted
+func (p *portworx) updatePortworxInstallStatus(
+	cluster *corev1.StorageCluster,
+	storageNodes []corev1.StorageNode,
+) {
+	if len(storageNodes) == 0 {
+		return
+	}
+	installCondition := util.GetStorageClusterCondition(cluster, pxutil.PortworxComponentName, corev1.ClusterConditionTypeInstall)
+	// Install only happens once on fresh install
+	if installCondition != nil && installCondition.Status == corev1.ClusterConditionStatusCompleted {
+		return
+	}
+
+	var inProgress bool
+	for _, storageNode := range storageNodes {
+		if storageNode.Status.Phase == string(corev1.NodeInitStatus) {
+			inProgress = true
```

Comment:
Instead of breaking here, we should use the number of nodes that have completed and the ones that are in progress and use them in the message of the condition. For example -  `Installation completed on 4/10 nodes, 6 nodes remaining.`